### PR TITLE
*: Add shuffle hot region scheduler

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
Mainly use this schedule in the test. Used to test whether TiKV will be affected under poor hotspot scheduling.

### What is changed and how it works?
 move hot region and transfer the leader to the newly added hot peer

### Check List 

Tests 

 - Unit test
 - Manual test

Code changes
 - Add an API interfaces

### CODE REVIEW

File: server/schedulers/scheduler_test.go [hunk 1]
```diff
@@ -272,3 +272,54 @@ func (s *testRejectLeaderSuite) TestRejectLeader(c *C) {
...
+	//|-----------|--------------|----------------|----------------|---------------|
+	//|     1     |       1      |        2       |       3        |      512KB    |
+	//|     2     |       1      |        3       |       4        |      512KB    |
+	//|     3     |       1      |        2       |       4        |      512KB    |
+	tc.AddLeaderRegionWithWriteInfo(1, 1, 512*1024*schedule.RegionHeartBeatReportInterval, 2, 3)
+	tc.AddLeaderRegionWithWriteInfo(2, 1, 512*1024*schedule.RegionHeartBeatReportInterval, 3, 4)
+	tc.AddLeaderRegionWithWriteInfo(3, 1, 512*1024*schedule.RegionHeartBeatReportInterval, 2, 4)
+	opt.HotRegionLowThreshold = 0
+
+	// try to get a operator
```
Comment (rleungx) [inline server/schedulers/scheduler_test.go]: an

File: server/schedulers/shuffle_hot_region.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+			if err != nil {
+				return nil, errors.WithStack(err)
+			}
+			limit = l
+		}
+		return newShuffleHotRegionScheduler(opController, limit), nil
+	})
+}
+
+type shuffleHotRegionScheduler struct {
```
Comment (rleungx) [inline server/schedulers/shuffle_hot_region.go]: Maybe we can add a comment to indicate this scheduler is for testing.

File: server/schedulers/shuffle_hot_region.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	case hotWriteRegionBalance:
+		s.stats.writeStatAsLeader = calcScore(cluster.RegionWriteStats(), cluster, core.LeaderKind)
+		return s.randomSchedule(cluster, s.stats.writeStatAsLeader)
+	}
+	return nil
+}
+
+func (s *shuffleHotRegionScheduler) randomSchedule(cluster schedule.Cluster, storeStats core.StoreHotRegionsStat) []*schedule.Operator {
+	for _, stats := range storeStats {
+		i := s.r.Intn(stats.RegionsStat.Len())
```
Comment (disksing) [inline server/schedulers/shuffle_hot_region.go]: Can `RegionStat` be empty?

Comment (nolouch) [reply server/schedulers/shuffle_hot_region.go]: `calcScore` make sure it is not empty.

Comment (disksing) [reply server/schedulers/shuffle_hot_region.go]: ok

**Pull request merged**